Modules/_zstd/compressor.c

```diff
@@ -195,12 +197,15 @@ _get_CDict(ZstdDict *self, int compressionLevel)
             goto error;
         }
 
-        /* Add PyCapsule object to self->c_dicts */
-        if (PyDict_SetItem(self->c_dicts, level, capsule) < 0) {
-            Py_DECREF(capsule);
+        /* Add PyCapsule object to self->c_dicts if not already inserted */
+        PyObject *capsule_dict;
+        int ret = PyDict_SetDefaultRef(self->c_dicts, level, capsule,
+                                       &capsule_dict);
+        Py_XDECREF(capsule_dict);
```

Comment:
```suggestion
        int ret = PyDict_SetDefaultRef(self->c_dicts, level, capsule, NULL);
```
You can pass NULL since `capsule_dict` is not needed